snapcraft/commands/discovery.py

```diff
@@ -66,7 +72,13 @@ def run(self, parsed_args):
                 snap_project = get_snap_project()
                 # Run this to trigger legacy behavior
                 yaml_data = process_yaml(snap_project.project_file)
-                project = Project.unmarshal(yaml_data)
+
+                # process yaml before unmarshalling the data
+                arch = get_host_architecture()
+                yaml_data_for_arch = apply_yaml(yaml_data, arch, arch)
+                extract_parse_info(yaml_data_for_arch)
```

Comment:
May this worth a comment why now `parse-info` needs to be extracted before unmarshalling the project?

Reply:
added a comment